cmake/PAL.cmake

```diff
@@ -236,17 +380,43 @@ function(ly_get_absolute_pal_filename out_name in_name)
 endfunction()
 
 function(ly_get_list_relative_pal_filename out_name in_name)
-    ly_get_absolute_pal_filename(abs_name ${in_name} ${ARGN})
+    message(DEPRECATION "ly_get_list_relative_pal_filename is being deprecated, change your code to use o3de_pal_dir instead.")
+    
+    # parent relative path is optional
+    if(${ARGC} GREATER 4)
+        set(parent_relative_path ${ARGV4})
+    endif()
+
+    # The Default object path for path is the LY_ROOT_FOLDER
+    cmake_path(SET object_path NORMALIZE "${LY_ROOT_FOLDER}")
+    if(${ARGC} GREATER 3)
+        # The user has supplied an object restricted path, the object path for consideration
+        cmake_path(SET object_path NORMALIZE ${ARGV3})
+    endif()
+
+    # The default restricted object path is O3DE_ENGINE_RESTRICTED_PATH
+    cmake_path(SET object_restricted_path NORMALIZE "${O3DE_ENGINE_RESTRICTED_PATH}")
+    if(${ARGC} GREATER 2)
+        # The user has supplied an object restricted path
+        cmake_path(SET object_restricted_path NORMALIZE ${ARGV2})
+    endif()
+
+    if(${ARGC} GREATER 4)
+        o3de_pal_dir(abs_name ${in_name} ${object_restricted_path} ${object_path} ${parent_relative_path})
+    else()
+        o3de_pal_dir(abs_name ${in_name} ${object_restricted_path} ${object_path})
+    endif()
```

Comment:
Instead of copy and pasting this code from the `ly_get_absolute_pal_filename` above, this should just still call that function